include/async_image_loader.h

```diff
@@ -0,0 +1,78 @@
+//
+// Created by Alexey Klimov on 13.01.2022.
+//
+
+#ifndef ASYNCIMAGELOADER_H_
+#define ASYNCIMAGELOADER_H_
+
+
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+
+#include <readerwriterqueue.h>
+
+
+using moodycamel::ReaderWriterQueue;
+
+class AsyncImageLoader {
+public:
+    explicit AsyncImageLoader(const char* folder, size_t start_frame,
+                              size_t last_frame, bool color = true):
+            mQueue(last_frame - start_frame)
+          , mDatasetFolder(folder)
+          , mIsColor(color)
+          , mThread([=]() {
+              for(auto i = start_frame; (i < last_frame) && finish; ++i) {
```

Comment:
no need to use auto here (simple type) change it for better readability -> size_t